pkg/process/checks/process_linux_test.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package checks
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"strings"
+	"testing"
+	"time"
+
+	wmdef "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	wmimpl "github.com/DataDog/datadog-agent/comp/core/workloadmeta/impl"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	probemocks "github.com/DataDog/datadog-agent/pkg/process/procutil/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestProcessesByPIDWLM tests processesByPID map creation when WLM collection is ON
+func TestProcessesByPIDWLM(t *testing.T) {
+	for _, tc := range []struct {
+		description  string
+		collectStats bool
+	}{
+		{
+			description:  "wlm collection ENABLED, with stats ENABLED",
+			collectStats: true,
+		},
+		{
+			description:  "wlm collection ENABLED, with stats DISABLED",
+			collectStats: false,
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			// INITIALIZATION
+			mockProbe := probemocks.NewProbe(t)
+			mockWLM := wmimpl.NewMockWLM(t)
+			mockConstantClock := constantMockClock(time.Now())
+			processCheck := &ProcessCheck{
+				wmeta:                   mockWLM,
+				probe:                   mockProbe,
+				useWLMProcessCollection: true,
+				clock:                   mockConstantClock,
+			}
+
+			// MOCKING
+			nowSeconds := mockConstantClock.Now().Unix()
+			proc1 := wlmProcessWithCreateTime(1, "git clone google.com", nowSeconds)
+			proc2 := wlmProcessWithCreateTime(2, "mine-bitcoins -all -x", nowSeconds-1)
+			proc3 := wlmProcessWithCreateTime(3, "datadog-agent --cfgpath datadog.conf", nowSeconds+2)
+			proc4 := wlmProcessWithCreateTime(4, "/bin/bash/usr/local/bin/cilium-agent-bpf-map-metrics.sh", nowSeconds-3)
+			procs := []*wmdef.Process{proc1, proc2, proc3, proc4}
+			statsByPid := make(map[int32]*procutil.Stats)
+			mockWLM.EXPECT().ListProcesses().Return(procs).Once()
```

Comment:
I think this could be replaced with `Set` calls.
If I’m not mistaken, the other use of the mock is in another file and just checks that ListProcesses wasn’t called.
If that’s the case, I think adding a new mock based on mockery isn’t worth the added complexity.